File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        final Map map = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            final int diff = target - nums[i];
            final Integer stored = (Integer)map.get(diff);
            if (stored != null) {
                return new int[]{stored, i};
            }
            map.put(nums[i], i);
        }
        return new int[]{0, 1};
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.kt
class Solution {
    fun search(nums: IntArray, target: Int): Int { // nums=[-1,0,2,4,6,8] target = 4
        var result = -1

        var l = 0
        var r = nums.size - 1

        while (l <= r) {
            val m = l + (r - l) / 2

            if (nums[m] == target) {
                result = m
                break
            } else if (nums[m] < target) {
                l = m + 1
            } else {
                r = m - 1
            }
        }

        return result
   }
}
